### DISCUSSION TITLE
How to power the board?

### DISCUSSION THREAD

**Discussion by TheS1ngularity** [klaasnicolaas/home-assistant-glow#79] [Q&A] [answered]
Hi guys, my meter is outside the house and I'm not sure how to power the board in that location :( Any tips are most welcomed

**Comment by bruvv** [comment, marked as answer]
With a usb 5v power adapter.

**Reply by TheS1ngularity** [reply]
I kinda got that, but where do I plug that adapter outside of the house? I don't have any power socket where the electrical panel is and I can't just take a 220V lead from the panel itself... do you have a ppwer socket next to your electrical panel?

**Reply by bruvv** [reply]
I have a server near my power meter that has a usb port that it is plugged into. Otherwise you can use a powerbank that will have to be charged once every 6 months or so :)? Cost only 10 euro on Ali these days.

**Reply by TheS1ngularity** [reply]
And your server is plugged in a power outlet next to the power meter?

**Reply by bruvv** [reply]
Correct

**Reply by nefelodamon** [reply]
What you could "try" to do is to get the sensor 3 pins to a longer cable extension (preferably multi-strand one to lower the resistance) and then the ESP board to the closest power socket. If its only a few meters you may get away with it?
Alternately you will have to bring power to the panel with an extension lead. 

Unfortunately the ESP will not run from a battery as it needs to be "awake" all the time to record the blinks. Any projects I saw with an ESP powered by battery was putting the ESP to a deep sleep and was waking for only a few milliseconds at a time to just send data.

**Reply by davisgoodman** [reply]
How far away from the ESP the photo-diode could be extended? Same here my meter is about 60 feet from the house, wondering if I could just extend the output of the Photo-Diode module to the ESP. As for powering the ESP I have a POE adaptor so worst case scenario will be to bring an Ethernet cable behind the pole where my meter is outside and use an IP67 case where the ESP will be

**Reply by CommittotheIndian** [reply]
> How far away from the ESP the photo-diode could be extended? Same here my meter is about 60 feet from the house, wondering if I could just extend the output of the Photo-Diode module to the ESP. As for powering the ESP I have a POE adaptor so worst case scenario will be to bring an Ethernet cable behind the pole where my meter is outside and use an IP67 case where the ESP will be

Did you ever try this? I'm wondering the same if you just lengthen the connections between the two. My only concern would be the delay in reading.

**Reply by davisgoodman** [reply]
@CommittotheIndian 

This is what I ended up doing...
![HydroMeterReader-3](https://user-images.githubusercontent.com/37666918/185713205-b5093461-e005-4fd8-a156-0a7716de033c.jpg)
![HydroMeterReader](https://user-images.githubusercontent.com/37666918/185713219-613ffe20-0ba7-4c3c-90f4-038a22285d6c.jpg)
![HydroMeterReader-2](https://user-images.githubusercontent.com/37666918/185713247-c24ed179-22c1-46db-91c8-89b68588df5f.jpg)

I get wifi and data though I'm still struggling with the pulses reading. I'm getting data but something is not working as it should. The power consumption is always in the same range whether or not I turn on heaters or the dryer. I'm going try a new photo-diode module because even changing the pulse rate doesn't make any difference. The results change but still doesn't vary when more power is consumed
